docker/scripts/make-lockfile.R

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env Rscript
+
+# This script is used to make a lockfile based on the current renv environment,
+# including scpcaTools dependencies and any additional packages specified by the user.
+# This means that it does _not_ look at what files are used by scripts in the project, except
+# for the package DESCRIPTION file.
+
+library(optparse)
+
+option_list <- list(
+  make_option(
+    c("-f", "--lockfile"),
+    type = "character",
+    default = NULL,
+    help = "The lockfile to create."
+  ),
+  make_option(
+    c("-p", "--packages"),
+    type = "character",
+    default = "",
+    help = paste(
+      "Additional packages to include in the lockfile beyond scpcaTools requirements.",
+      "Comma, semicolon, or space separated."
+    )
+  )
+)
+
+opts <- parse_args(OptionParser(option_list = option_list))
+
+# change to project root to get renv (renv::load does not set bioc mirrors)
+current_dir <- getwd()
+setwd(here::here())
+source(".Rprofile")
+setwd(current_dir)
+
+# get dependencies of scpcaTools
+renv::snapshot(lockfile = opts$lockfile, type = "explicit")
```

Comment:
What's the output of this? If I run a line like this locally, I get a prompt to update packages?

Reply:
The output of this should be a new lockfile at `opts$lockfile`. Not sure why you are getting a prompt to update... That seems odd. Maybe it prompts to update if your packages are not up-to-date first?

Reply:
Yes that's definitely it. I guess if you make sure to run `renv::snapshot` on the main lockfile before running this script then it should work correctly.